README.md

```diff
@@ -4,64 +4,170 @@ Rubyric is an online tool in which you can create courses and assignments,
+```
+
+If you have a different Bundler version which is not compatible with the Rubyric 
+ecosystem, you must uninstall it and install a compatible version. The command
+for installing the bundler is similar to the one below. However, you can choose 
+the most suitable [version](https://rubygems.org/gems/bundler/versions) for you.
+
+```sh
+gem install bundler -v 1.17.3
+```
+
+### Database
+
+Rubyric uses a PostgreSQL database. Therefore, you must install PostgreSQL and 
+create a user.
+
 ```sh
 sudo apt-get install postgresql
 sudo -u postgres createuser --interactive
 ```
-Rubyric requires pdfinfo, ghostscript and libpq-dev to work
+
+### Additional packages
+
+Rubyric also requires pdfinfo, ghostscript and libpq-dev to work.
 ```sh
 sudo apt-get install poppler-utils ghostscript libpq-dev
 ```
 
-## Install Rubyric
+## Download Rubyric and Install Gems
+
+### Clone the Rubyric project
+
+Now it is time to clone the Rubyric project and install the required gems. First,
+clone the project from [Github](https://github.com/Aalto-LeTech/rubyric). You can 
+use either SSH or HTTPS.
+
+```sh
+git clone git@github.com:Aalto-LeTech/rubyric.git && cd rubyric
+```
+
+or
+
+```sh
+git clone https://github.com/Aalto-LeTech/rubyric.git && cd rubyric
+```
 
 ### Install gems
+
+Once you have a copy of the Rubyric project, you can proceed to install the 
+required gems.
+
 ```sh
 bundle install
 ```
 
-### Configure
+**Side note:** Some packages will require manual installation, and some other 
+will require some manual intervention related to compatibility issues. You can
+find some of the most recurrent problems in the 
+[Troubleshooting section](https://github.com/Aalto-LeTech/rubyric/blob/master/README.md#troubleshooting)
+
+
+## Configure Rubyric
+
+### Local configuration
+
+Now that you have installed  the required software and tools, you must configure
+your Rubyric project to run in your computer.
+
 ```sh
 cp config/initializers/secret_token.rb.base config/initializers/secret_token.rb
 cp config/initializers/settings.rb.base config/initializers/settings.rb
 cp config/database.yml.base config/database.yml
 ```
 
-### Create database, and put password and username to config/database.yml
+### Create a database and username
+
+The firs step is to create the database and the rails user
 ```sh
-sudo -u postgres createdb -O my_username rubyric
+sudo -u postgres createdb -O rails rubyric
 ```
 
-### Initialize database
+Now, we will create a user called "rails" with password "rails". This particular 
+user is used for testing purposes.
+
 ```sh
-rails db:setup
+sudo -u postgres psql
 ```
+The PostgreSQL console is indicated by `postgres=#` prompt. At the PostgreSQL 
+console enter the following command, which will assist you in assigning the 
+password to the "rails" user.
 
-### Start server
 ```sh
-bin/delayed_job start
-rails server
+postgres=# \password rails
 ```
+Enter the desired password (in this case, your password should be "rails") at 
```

Comment:
I would recommend using ident authentication with database servers (postgresql and mariadb) for better security. Hence the `createdb -O $USER rubyric`

Reply:
I am using this user for development purposes [database.yml.base](https://github.com/Aalto-LeTech/rubyric/blob/45168fb2fa096adb4a2d7d7edbf0c947488f473c/config/database.yml.base#L3). Nevertheless, I will add one more line below to mention about this "ident authentication"

Reply:
The point is, as long as the database is in the same machine, there is no point to use password. However, if the database is in another machine (e.g. container), then ident authentication is not possible.

Reply:
Yes, maybe there is no point in using a password, but I am documenting what exists. In the [current implementation](https://github.com/Aalto-LeTech/rubyric/blob/45168fb2fa096adb4a2d7d7edbf0c947488f473c/config/database.yml.base#L2), they set the username and password. Thus my intention with this PR is not to modify the code but to extend the current documentation, to help the user run Rubyric locally.

I will rephrase the last paragraph I added, and I think that will be enough for documenting the current source code.